### PR TITLE
Many fixes and changes

### PR DESCRIPTION
**Pull request by DarthRogue**
I've made many changes and gotten things going in the right direction.  There is still a laundry list of things to be done.  AI are still not staying after spawning.  The server is cleaning them up right away.  It likely has to do with the fact that units must be assigned an owner when they are spawned, and I don't believe this is done in the current code.

We also need to get the vehicle publishing working, but with the new hive system I'm not up to speed on how to make it happen.

I've taken this about as far as I can with my current skills.  Many communities are waiting on this AI pack and I feel that at this point for me to work on it any more would be holding back progress.  I can tinker with it some but I think you should probably dive into it from here.  I'm happy to receive some guidance as to how to proceed if you find some stuff that you think I can fix with a little instruction.

### CODE REVIEW

**Comment by nerdalertdk** [comment]
can see the backpack an vest names are wrong, can fixit when the pull is approved

**Comment by nerdalertdk** [comment]
this should stop the clean up
```
//Stops the AI from being cleaned up
_ai1 setVariable["LASTLOGOUT_EPOCH",1000000000000];
_ai1 setVariable["LAST_CHECK",1000000000000];
```

File: @WAI/compile/mission_winorfail.sqf [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+
+		if (typeOf(_crate) in (crates_large + crates_medium + crates_small)) then {
+
+			if(wai_crates_smoke && sunOrMoon == 1) then {
+				_marker = "smokeShellPurple" createVehicle getPosATL _crate;
+				_marker setPosATL (getPosATL _crate);
+				_marker attachTo [_crate,[0,0,0]];
+			};
+
+			if (wai_crates_flares && sunOrMoon != 1) then {
```
Comment (nerdalertdk) [inline @WAI/compile/mission_winorfail.sqf]: Will not work in arma3 / epoch, its a arma 2 dayz thing, possible replacement "FlareWhite_F"

File: remote/remote.sqf [hunk 2]
```diff
@@ -8,7 +8,7 @@ if(!isDedicated) then {
 		_message 	= _this select 1;
 		
 		call {
-			if(_type == "radio")		exitWith { if(player hasWeapon "ItemRadio") then { systemChat _message; }; };
+			if(_type == "radio")		exitWith { if(player hasWeapon "EpochRadio0") then { systemChat _message; }; };
```
Comment (nerdalertdk) [inline remote/remote.sqf]:
```
_Radios = ["EpochRadio0","EpochRadio1","EpochRadio2","EpochRadio3","EpochRadio4","EpochRadio5","EpochRadio6","EpochRadio7","EpochRadio8","EpochRadio9"];
if(_type == "radio")		exitWith { if(_Radios in (assignedItems  player)) then { systemChat _message; }; };
```

untested

File: WAI/config.sqf [hunk 3]
```diff
@@ -141,18 +141,18 @@ if(isServer) then {
...
 
 		// Dynamic box array
-		crates_large				= ["Cargo_Container"];  //MAY NOT HOLD ALL NEEDED ITEMS
-		crates_medium				= ["C_supplyCrate_F"];  //MAY NOT HOLD ALL NEEDED ITEMS
-		crates_small				= ["Pelican_EPOCH"];  //MAY NOT HOLD ALL NEEDED ITEMS
+		crates_large				= ["Cargo_Container"];  
+		crates_medium				= ["C_supplyCrate_F"];  
+		crates_small				= ["Pelican_EPOCH"];  
 
 		crate_weapons_buildables	= [["ChainSaw","CSGAS"],["ChainSawB","CSGAS"],["ChainSawG","CSGAS"],["ChainSawP","CSGAS"],["ChainSawR","CSGAS"]];
```
Comment (nerdalertdk) [inline WAI/config.sqf]: crate_weapons_buildables	= [["MeleeSledge","swing"],["Hatchet","swing"],["ChainSawG","CSGAS"],["ChainSawP","CSGAS"],["ChainSawR","CSGAS"]];

File: @WAI/fn_init.sqf [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+	dynamic_crate 			= compile preprocessFileLineNumbers "\x\addons\WAI\compile\dynamic_crate.sqf";
+
+	ai_monitor				= compile preprocessFileLineNumbers "\x\addons\WAI\compile\ai_monitor.sqf";
+	vehicle_monitor			= compile preprocessFileLineNumbers "\x\addons\WAI\compile\vehicle_monitor.sqf";
+	find_position			= compile preprocessFileLineNumbers "\x\addons\WAI\compile\find_position.sqf";
+	load_ammo				= compile preprocessFileLineNumbers "\x\addons\WAI\compile\load_ammo.sqf";
+
+	call 					compile preprocessFileLineNumbers "\x\addons\WAI\compile\functions.sqf";
+
+	if(isNil("DZMSInstalled")) then {
```
Comment (nerdalertdk) [inline @WAI/fn_init.sqf]: leave this out ?

**Comment by DarthRogue** [comment]
Recommended changes made.

File: remote/remote.sqf [hunk 5]
```diff
@@ -6,9 +6,9 @@ if(!isDedicated) then {
 	
 		_type 		= _this select 0;
 		_message 	= _this select 1;
-		
+		_Radios = ["EpochRadio0","EpochRadio1","EpochRadio2","EpochRadio3","EpochRadio4","EpochRadio5","EpochRadio6","EpochRadio7","EpochRadio8","EpochRadio9"];
 		call {
-			if(_type == "radio")		exitWith { if(player hasWeapon "ItemRadio") then { systemChat _message; }; };
+			if(_type == "radio")		exitWith { if(player hasWeapon "_Radios") then { systemChat _message; }; };
```
Comment (nerdalertdk) [inline remote/remote.sqf]: HasWeapon need a object not a array
https://community.bistudio.com/wiki/hasWeapon

**Comment by f3cuk** [comment]
Hey guys, i cannot auto merge these changes. Please make sure they are auto-mergeable.

**Comment by f3cuk** [comment]
Okay done now from the command line. Please make sure you are always up-to-date before pushing :+1:

File: remote/remote.sqf (hunk 5 above)
Comment (f3cuk) [inline remote/remote.sqf]: Use this instead
```cpp
if("_Radios" in (items player + assignedItems player))
```

**Pull request merged**